src/modules/SearchPage/components/Cards/index.tsx

```diff
@@ -14,96 +14,83 @@ type Props = {
   isLoading: boolean;
 };
 
-type State = {
-  selectedType?: PokemonTypeDetails;
-  selectedMove?: PokemonMoveDetails;
-  selectedPokemon?: PokemonDetails;
-  isModalOpened: boolean;
-};
-
-export class Cards extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      isModalOpened: false,
-      selectedPokemon: undefined,
-      selectedMove: undefined,
-      selectedType: undefined,
-    };
-  }
+export const Cards = ({ activeTab, isLoading, moves, pokemons, types }: Props) => {
+  const [selectedPokemon, setSelectedPokemon] = useState<PokemonDetails>();
+  const [selectedType, setSelectedType] = useState<PokemonTypeDetails>();
+  const [selectedMove, setSelectedMove] = useState<PokemonMoveDetails>();
+  const [isModalOpened, setIsModalOpened] = useState(false);
 
-  handlePokemonSelect = (selectedPokemon: PokemonDetails) => this.setState({ selectedPokemon });
-  handleMoveSelect = (selectedMove: PokemonMoveDetails) => this.setState({ selectedMove });
-  handleTypeSelect = (selectedType: PokemonTypeDetails) => this.setState({ selectedType });
-  handleModalOpened = (isModalOpened: boolean) => this.setState({ isModalOpened });
+  const handleCloseModal = useCallback(() => {
+    setSelectedPokemon(undefined);
+    setSelectedType(undefined);
+    setSelectedMove(undefined);
```

Comment:
https://stackoverflow.com/questions/70554259/why-should-we-not-explicitly-assign-undefined-to-a-variable-in-javascript

Use null as default instead.

Reply:
done